x-pack/plugins/apm/public/components/app/TraceOverview/TraceList.tsx

```diff
@@ -75,19 +73,19 @@ const TRACE_COLUMNS = [
     width: '20%',
     align: 'right',
     sortable: true,
-    render: (value: number) => <ImpactBar value={value} />
+    render: (value: number) => <ImpactBar value={value} range={items} />
   }
 ];
 
 interface Props {
-  items: Array<StringMap<any>>;
+  items: ImpactItem[];
   noItemsMessage: any;
 }
 
 export function TraceList({ items = [], noItemsMessage }: Props) {
   return (
     <ManagedTable
-      columns={TRACE_COLUMNS}
+      columns={getTraceColumns({ items })}
```

Comment:
Probably not a big issue, but before `TRACE_COLUMNS ` was defined outside the render function, and thus `ManagedTable` would not re-render whenever `TraceList` re-rendered. Now `getTraceColumns` returns a new object every time `TraceList` re-renders , which will cause `ManagedTable` to re-render also.